centreon/src/Centreon/Domain/MonitoringServer/UseCase/GenerateConfiguration.php

```diff
@@ -69,7 +61,17 @@ public function __construct(
     public function execute(int $monitoringServerId): void
     {
         try {
-            $monitoringServer = $this->monitoringServerRepository->findServer($monitoringServerId);
+            if (! $this->contact->isAdmin()) {
+                $accessGroups = $this->readAccessGroupRepositoryInterface->findByContact($this->contact);
+
+                $monitoringServer = $this->monitoringServerRepository->findServerByIdAndAccessGroups(
```

Comment:
The best solution would be to create an "exists" method that could contain a nullable list of access groups.
If the accessGroups list is null, no check with ACL, if accessGroups is an array, we check with joins